### PR TITLE
[TT-11598, DX-1191]guide users to extend official image with python

### PR DESCRIPTION
**Pull request by jeffy-mathew**
### For internal users - Please add a Jira DX PR ticket to the subject!
https://tyktech.atlassian.net/browse/TT-11598
<br>

---

### Preview Link

[preview](https://deploy-preview-4245--tyk-docs.netlify.app/docs/nightly/plugins/supported-languages/rich-plugins/python/python)

### Description

<br>

#### Screenshots (if appropriate)
<br>

## Checklist

- [x] I have added a **preview link** to the PR description.
- [x] I have [reviewed the guidelines](https://github.com/TykTechnologies/tyk-docs/blob/master/CONTRIBUTING.md) for contributing to this repository.
- [x] I have [read the technical guidelines](https://github.com/TykTechnologies/tyk-docs/blob/master/CONTRIBUTING-TECHNICAL-GUIDE.md) for contributing to this repository.
- [x] Make sure you have started *your change* off *our latest `master`*.
- [x] I **labelled** the PR

___

## **Type**
enhancement, documentation

___

## **Description**
- Added detailed instructions for setting up Python development environment for Tyk plugins across different platforms including Docker.
- Introduced a Dockerfile example to help users extend the official Tyk Gateway Docker image with Python, ensuring compatibility with Python plugins in Tyk Gateway version `v5.3.0` and above.

___

## **Changes walkthrough**
<table><thead><tr><th></th><th align="left">Relevant files</th></tr></thead><tbody><tr><td><strong>Enhancement
</strong></td><td><table>
<tr>
  <td>
    <details>
      <summary><strong>python.md</strong><dd><code>Extend Official Tyk Gateway Docker Image with Python for Plugin </code><br><code>Development</code></dd></summary>
<hr>

tyk-docs/content/plugins/supported-languages/rich-plugins/python/python.md
<li>Added instructions for installing Python development packages on <br>Ubuntu/Debian, Red Hat or CentOS, and Docker.<br> <li> Provided a Dockerfile example to extend the official Tyk Gateway <br>Docker image with Python for plugin development.<br> <li> Updated the installation guide to include a tabbed interface for <br>different operating systems and Docker.<br>

</details>
    

  </td>
  <td><a href="https://github.com/TykTechnologies/tyk-docs/pull/4245/files#diff-995d274eb8480c5d83f5d02e9e5958c5400b04c8a981b5834dd9634b81c3adcd">+38/-3</a>&nbsp; &nbsp; </td>
</tr>                    
</table></td></tr></tr></tbody></table>

___

> ✨ **PR-Agent usage**:
>Comment `/help` on the PR to get a list of all available PR-Agent tools and their descriptions

### CODE REVIEW

File: tyk-docs/content/plugins/supported-languages/rich-plugins/python/python.md [hunk 1]
```diff
@@ -20,14 +20,54 @@ Since v2.9, Tyk supports any currently stable [Python 3.x version](https://www.p
...
 ### Install the Python development packages
 
-If you're using Ubuntu/Debian:
+{{< tabs_start >}}
 
+{{< tab_start "Docker" >}}
+Starting from Tyk Gateway version `v5.3.0`, Python is no longer bundled with the official Tyk Gateway Docker image. 
+However, Tyk Gateway continues to support Python plugins functionality. 
+
+If you wish to use Python plugins using docker, you can extend the official Tyk Gateway Docker image by adding Python to it.
```
Comment (andyo-tyk) [inline tyk-docs/content/plugins/supported-languages/rich-plugins/python/python.md]:
```suggestion
If you wish to use Python plugins using Docker, you can extend the official Tyk Gateway Docker image by adding Python to it. 
```

File: tyk-docs/content/plugins/supported-languages/rich-plugins/python/python.md [hunk 2]
```diff
@@ -20,14 +20,54 @@ Since v2.9, Tyk supports any currently stable [Python 3.x version](https://www.p
...
-If you're using Ubuntu/Debian:
+{{< tabs_start >}}
 
+{{< tab_start "Docker" >}}
+Starting from Tyk Gateway version `v5.3.0`, Python is no longer bundled with the official Tyk Gateway Docker image. 
+However, Tyk Gateway continues to support Python plugins functionality. 
+
+If you wish to use Python plugins using docker, you can extend the official Tyk Gateway Docker image by adding Python to it. 
+
+An example Dockerfile that extends official Tyk Gateway image and installs python and required modules is shown below.
```
Comment (andyo-tyk) [inline tyk-docs/content/plugins/supported-languages/rich-plugins/python/python.md]:
```suggestion
This example Dockerfile extends the official Tyk Gateway image to support Python plugins by installing python and the required modules:
```

File: tyk-docs/content/plugins/supported-languages/rich-plugins/python/python.md [hunk 3]
```diff
@@ -20,14 +20,54 @@ Since v2.9, Tyk supports any currently stable [Python 3.x version](https://www.p
...
+
+ENV PORT=8080
+
+WORKDIR /opt/tyk-gateway/
+
+ENTRYPOINT ["/opt/tyk-gateway/tyk" ]
+CMD [ "--conf=/opt/tyk-gateway/tyk.conf" ]
+```
+
+Run docker build using this Dockerfile, provide Tyk Gateway image that you would like to extend as build argument `BASE_IMAGE`.
```
Comment (andyo-tyk) [inline tyk-docs/content/plugins/supported-languages/rich-plugins/python/python.md]:
```suggestion
To use this, you simply run `docker build` with this Dockerfile, providing the Tyk Gateway image that you would like to extend as build argument `BASE_IMAGE`.
```

File: tyk-docs/content/plugins/supported-languages/rich-plugins/python/python.md [hunk 4]
```diff
@@ -20,14 +20,54 @@ Since v2.9, Tyk supports any currently stable [Python 3.x version](https://www.p
...
+For eg., the following command would extend Tyk Gateway `v5.3.0` by installing Python and related dependencies for Python plugins and builds the image `tyk-gateway-python:v5.3.0`.
```
Comment (andyo-tyk) [inline tyk-docs/content/plugins/supported-languages/rich-plugins/python/python.md]:
```suggestion
As an example, this command will extend Tyk Gateway `v5.3.0` to support Python plugins, generating the image `tyk-gateway-python:v5.3.0`:
```

File: tyk-docs/content/plugins/supported-languages/rich-plugins/python/python.md [hunk 5]
```diff
@@ -20,14 +20,54 @@ Since v2.9, Tyk supports any currently stable [Python 3.x version](https://www.p
...
+
+{{< tab_start "Ubuntu/Debian" >}}
 ```apt
 apt install python3 python3-dev python3-pip build-essential
 ```
 
-If you're using Red Hat or CentOS:
+### Install the Required Python Modules
+
+Make sure that "pip" is now available in your system, it should be typically available as "pip", "pip3" or "pipX.X" (where X.X represents the Python version):
```
Comment (andyo-tyk) [inline tyk-docs/content/plugins/supported-languages/rich-plugins/python/python.md]:
```suggestion
Make sure that "pip" is available in your system, it should be typically available as "pip", "pip3" or "pipX.X" (where X.X represents the Python version):
```

File: tyk-docs/content/plugins/supported-languages/rich-plugins/python/python.md [hunk 6]
```diff
@@ -20,14 +20,54 @@ Since v2.9, Tyk supports any currently stable [Python 3.x version](https://www.p
 
 ### Install the Python development packages
 
-If you're using Ubuntu/Debian:
+{{< tabs_start >}}
 
+{{< tab_start "Docker" >}}
+Starting from Tyk Gateway version `v5.3.0`, Python is no longer bundled with the official Tyk Gateway Docker image.
```
Comment (andyo-tyk) [inline tyk-docs/content/plugins/supported-languages/rich-plugins/python/python.md]:
```suggestion
{{< note success >}}
**Note**  

Starting from Tyk Gateway version `v5.3.0`, Python is no longer bundled with the official Tyk Gateway Docker image so you must extend the image to add support for Python plugins.
{{< /note >}}
```

File: tyk-docs/content/plugins/supported-languages/rich-plugins/python/python.md [hunk 7]
```diff
@@ -20,14 +20,54 @@ Since v2.9, Tyk supports any currently stable [Python 3.x version](https://www.p
 
 ### Install the Python development packages
 
-If you're using Ubuntu/Debian:
+{{< tabs_start >}}
 
+{{< tab_start "Docker" >}}
+Starting from Tyk Gateway version `v5.3.0`, Python is no longer bundled with the official Tyk Gateway Docker image. 
+However, Tyk Gateway continues to support Python plugins functionality.
```
Comment (andyo-tyk) [inline tyk-docs/content/plugins/supported-languages/rich-plugins/python/python.md]:
```suggestion
```

**Review by andyo-tyk — CHANGES_REQUESTED** [review]
Added some suggestions for wording improvements

**Comment by dcs3spp** [comment]
Thanks for updating the content @jeffy-mathew , to enable review of PR could you create a [Jira ticket ](https://tyktech.atlassian.net/browse/DX)in DX space and add the DX ticket ID to the PR title and description

**Review by titpetric — APPROVED** [review]
technical review lgtm

**Review by dcs3spp — COMMENTED** [review]
Is it worth adding the docker buildx equivalent also?

_docker buildx build --build-arg BASE_IMAGE=tykio/tyk-gateway:v5.3.0 -t tyk-gateway-python:v5.3.0 ._

**Review by dcs3spp — APPROVED** [review]
@jeffy-mathew PR LGTM and approved, left comment regarding _docker buildx_ 

@andyo-tyk if all changes are ok can you approve and then I will merge to main branch ready for release

**Review by andyo-tyk — APPROVED** [review]
LGTM, thanks.

**Pull request merged**